src/layers/mlp_llama.h

```diff
@@ -41,16 +41,41 @@ class LlamaMLP : public SingletonBase<LlamaMLP<WeiT>> {
     LlamaMLP(DecoderContext *ctx) {}
 
     // The inerface is for PyTorch, thus the weights are already transposed
-    void setWeights(DecoderContext *ctx, std::vector<float *> &params, bool trans = true) {
+    void setWeights(DecoderContext *ctx, std::vector<void *> &params, bool trans = true, int type = 0) {
+        // set float32 weights
+        if (type == 0) {
+            // Refer to CommonDecoder for parameters order
+            const float *gateW = (const float *)params[0];
+            const float *upW = (const float *)params[2];
+            const float *normW = (const float *)params[4];
+            const float *downW = (const float *)params[6];
+            setWeights(ctx, gateW, nullptr, nullptr, upW, nullptr, nullptr, normW, downW, nullptr, nullptr, trans);
+        }
+        // set int8 weights
+        else if (type == 1) {
+            // Refer to CommonDecoder for parameters order
+            const int8_t *gateW = (const int8_t *)params[0];
+            const float *gateS = (const float *)params[1];
+            const float *gateZ = (const float *)params[2];
+            const int8_t *upW = (const int8_t *)params[4];
+            const float *upS = (const float *)params[5];
+            const float *upZ = (const float *)params[6];
+            const float *normW = (const float *)params[8];
+            const int8_t *downW = (const int8_t *)params[10];
+            const float *downS = (const float *)params[11];
+            const float *downZ = (const float *)params[12];
+            setWeights(ctx, gateW, gateS, gateZ, upW, upS, upZ, normW, downW, downS, downZ, trans);
+        }
+    }
+
+    // SrcT: float or int8_t
+    template <typename SrcT>
```

Comment:
SrcT means WeiT?

Reply:
SrcT is the source datatype of weight, convert to WeiT through MMHelper::convertWeight